AI/VCAI/Goals/Build.cpp

```diff
@@ -304,13 +310,48 @@ Tasks::TaskList Build::getTasks()
 		developmentInfos.push_back(developmentInfo);
 	}
 
+	requiredResources -= availableResources;
+	requiredResources.positive();
+
 	logAi->trace("daily income: %s", dailyIncome.toString());
 	logAi->trace("resources required to develop towns now: %s, total: %s",
 		requiredResources.toString(),
 		totalDevelopmentCost.toString());
 
 	//TODO: we need to try enforce capturing resources of particular type if we need them.
+	Tasks::TaskList resourceCapture;
+
+	for (int resType = Res::ERes::WOOD; resType <= Res::ERes::MITHRIL; resType++) {
+		if (resType == Res::ERes::GOLD) {
+			continue;
+		}
+
+		double daysToGetResource = requiredResources[resType] / (dailyIncome[resType] + 0.001); // avoid zero divide
+		double priority = resType == std::max(1.0, daysToGetResource / 10);
+
+		if (priority < 0.5) {
+			continue;
+		}
+
+		std::vector<const CGObjectInstance*> interestingObjects;
 
+		vstd::copy_if(ai->visitableObjs, std::back_inserter(interestingObjects), [&](const CGObjectInstance* obj) -> bool {
+			auto id = obj->ID;
+			auto mineOrResource = id == Obj::MINE || id == Obj::ABANDONED_MINE || id == Obj::RESOURCE;
+
+			return mineOrResource && obj->subID == resType;
+		});
+
```

Comment:
I believe there should still be intermediate step "gather resources" between "build" and "capture mine". Resources can also be generated via trading, and this functionality is already handled.